src/cpu/x64/jit_generator.hpp

```diff
@@ -182,6 +184,91 @@ class jit_generator : public Xbyak::CodeGenerator, public c_compatible {
 
     inline size_t get_size_of_abi_save_regs() { return size_of_abi_save_regs; }
 
+    using Xbyak::CodeGenerator::push;
+    using Xbyak::CodeGenerator::pop;
+
+    inline void push(const Xbyak::Xmm &xmm) {
+        if (xmm.isXMM()) {
+            sub(rsp, xmm_len);
+            uni_vmovdqu(ptr[rsp], xmm);
+        } else if (xmm.isYMM()) {
+            sub(rsp, ymm_len);
+            uni_vmovdqu(ptr[rsp], Xbyak::Ymm{xmm.getIdx()});
+        } else if (xmm.isZMM()) {
+            sub(rsp, zmm_len);
+            uni_vmovdqu(ptr[rsp], Xbyak::Zmm{xmm.getIdx()});
+        }
+    }
+
+    inline void push(const std::vector<Xbyak::Xmm> &xmms) {
+        std::vector<std::function<void()>> deferred_movs{};
+        size_t offset = 0;
+        for (size_t i = 0; i < xmms.size(); ++i) {
+            const auto& xmm = xmms[i];
+            if (xmm.isXMM()) {
+                deferred_movs.emplace_back([&]() {
+                    uni_vmovdqu(ptr[rsp + offset], xmm);
```

Comment:
Maybe, 'offset' should be captured by value here?

Reply:
Fixed

Reply:
This one may resolved now (I don't have rights to resolve comments)